packages/@aws-cdk/aws-cloudfront/lib/origin_access_identity.ts

```diff
@@ -0,0 +1,126 @@
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/core');
+import { CfnCloudFrontOriginAccessIdentity } from './cloudfront.generated';
+
+export interface CloudFrontOriginAccessIdentityProps {
+  /**
+   * Any comments you want to include about the origin access identity.
+   *
+   * @default "Allows CloudFront to reach to the bucket"
+   */
+  readonly comment?: string;
+}
+
+export interface ICloudFrontOriginAccessIdentity extends cdk.IResource, iam.IGrantable  {
+  /**
+   * The Origin Access Identity Name
+   */
+  readonly name: string;
+}
+
+export abstract class BaseCloudFrontOriginAccessIdentity extends cdk.Resource {
+  /**
+   * The Origin Access Identity Name (physical id)
+   */
+  public abstract readonly name: string;
+  /**
+   * Derived principal value for bucket access
+   */
+  public abstract readonly grantPrincipal: iam.IPrincipal;
+
+  /**
+   * The Arn to include in S3 bucket policy to allow CloudFront access
+   */
+  protected arn(): string {
+    return cdk.Stack.of(this).formatArn(
+      {
+        service: "iam",
+        region: "", // global
+        account: "cloudfront",
+        resource: "user",
+        resourceName: `CloudFront Origin Access Identity ${this.name}`
+      }
+    );
+  }
+}
+
+export interface CloudFrontOriginAccessIdentityAttributes {
+  /**
+   * The Origin Access Identity Name
+   */
+  readonly name: string;
+}
+
+/**
+ * An origin access identity is a special CloudFront user that you can
+ * associate with Amazon S3 origins, so that you can secure all or just some of
+ * your Amazon S3 content.
+ */
+export class CloudFrontOriginAccessIdentity extends BaseCloudFrontOriginAccessIdentity implements ICloudFrontOriginAccessIdentity {
+  public static fromCloudFrontOriginAccessIdentityAttributes(
```

Comment:
This method is not needed

Reply:
I thought this is a convention - removed it